### PR TITLE
First integration of Pixel Alignment of large structures in PCL (74X)

### PR DESCRIPTION
**Pull request by cerminar**
This PR integrates the alignment of the pixel large structures as a new PCL workflow for Tier0.
There is no code change to any production workflow, only the alca configurations are affected by the changes.

The new workflow has been integrated in the relval 1001 (which now has one more step).

### CODE REVIEW

**Comment by mmusich** [comment]
please test

File: Alignment/CommonAlignmentProducer/python/ALCARECOPromptCalibProdSiPixelAli_cff.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	applyChargeCheck = False,
+	minHitChargeStrip = 30.,
+)
+#Special option for PCL
+SiPixelAliTrackSelector.minHitsPerSubDet.inPIXEL = 2
+
+
+# Ingredient: SiPixelAliTrackRefitter0
+# refitting
+from RecoTracker.TrackProducer.TrackRefitters_cff import *
```
Comment (mmusich) [inline Alignment/CommonAlignmentProducer/python/ALCARECOPromptCalibProdSiPixelAli_cff.py]: @cerminar this inclusion here is causing the issues in the relval tests. There is duplication of definition of the `SiPixelClusters`. You might want to follow the recipe I used here https://github.com/cms-sw/cmssw/pull/9104/files to fix the SiStripAPVGain PCL

**Comment by cerminar** [comment]
@mmusich thanks for the suggestion. I implemented it. I still don't understand why this happens only in certain scenarios but I guess we have to leave with it.

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
trying to understand why it happens, maybe will save us some issues in the future

File: Alignment/CommonAlignmentProducer/test/TrackerAPForPCL/alignment_mille.py [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+#process.load("Configuration.StandardSequences.MagneticField_0T_cff")  ## FOR 0T
+
+#-- Load geometry
+process.load("Configuration.Geometry.GeometryIdeal_cff")
+
+#-- Global Tag
+process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+process.GlobalTag.connect = "frontier://FrontierProd/CMS_COND_31X_GLOBALTAG"
+process.GlobalTag.globaltag = "GR_P_V49::All"
+
```
Comment (mmusich) [inline Alignment/CommonAlignmentProducer/test/TrackerAPForPCL/alignment_mille.py]: @cerminar even if it is a test, maybe better to clean condDBv1 stuff around

**Comment by mmusich** [comment]
technically it works, though the pede dump looks suspicious.
Can please @frmeier @mschrode @ckleinw @chmartin please have a look and report?
The output of the workflow can be found here: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10452/191587/runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4/

**Comment by chmartin** [comment]
Hey,

So things seemed to have been called correctly but Pede did not try to minimize. My first guess is that there were not enough records. However the error explicitly says no free parameters, but a first pass at the configuration things are setup correctly. 

My understanding is that this error is thrown when the residuals passed cannot constrain the parameters asked for. So I think the code itself is working fine.

Chris

**Comment by mmusich** [comment]
@chmartin, thanks. The test is run over 10 events, so it's very likely there are not enough records to constrain anything.

**Comment by chmartin** [comment]
Hey,

I am currently running more robust tests on this branch to compare configuration/output to other workflows. Tomorrow I will know more about these tests, but the software itself looks to be great.

**Comment by cerminar** [comment]
@chmartin @mmusich this PR has nothing to do with the physics performance of the calibration algorithm and the discussion about the second doesn't belong to here.
If this is not integrated all the development will soon be lost because not maintainable anymore in private branches.

**Comment by frmeier** [comment]
+1
To me, the output looks compatible with an alignment with insufficient tracks. Better integrate and test it later than lost and/or unmaintainable code. Performance can follow later (and may require another PR...)
Just my 2 cents.

**Comment by mmusich** [comment]
@cerminar @cerminar if this PR causes regressions in the standard (non PCL) workflow this will imply substantial changes in the physics performance. I just want to avoid that. Once this is well demonstrated I have no problem to approve it (by the way you can still sign and override any decision from alca)

**Comment by davidlange6** [comment]
@cerminar @mmusich - is this PR needed?

**Comment by mmusich** [comment]
+1
@davidlange6 yes, since it has been already merged in 75X, 76X. It still needs the backport of #11255

**Comment by ggovi** [comment]
+1

**Comment by davidlange6** [comment]
@mmusich - is there a backport of #11255? I understand from you that this PR waits on that?
@cerminar

**Comment by mmusich** [comment]
@davidlange6 not that I am aware of, but I can prepare one if @cerminar cannot

**Comment by cerminar** [comment]
please test

**Comment by cerminar** [comment]
@davidlange6 and @mmusich: this should now be the state of the art...
Thanks,
G

**Comment by mmusich** [comment]
@cmsbuild please acknowledge changes

**Comment by mmusich** [comment]
+1 
contains also backport of #11255

**Comment by mmusich** [comment]
Hello,@danduggan, @boudoul, @franzoni, @fabozzi, @srimanob, @ggovi, 
can we have this signed again?
Thanks!
M.

**Comment by fabozzi** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by diguida** [comment]
@davidlange6 can this be integrated now, as discussed at the ORP?

**Pull request merged**